src/QsCompiler/Core/TransformationOptions.fs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.Transformations.Core
+
+
+/// Used to configure the behavior of the default implementations for transformations.
+type TransformationOptions = {
+    /// Disables the transformation at the transformation root, 
+    /// meaning the transformation won't recur into leaf nodes or subnodes. 
+    Disable : bool
```

Comment:
Also, I have a slight preference for booleans named positively, so this could possibly be renamed to Enabled and invert all the logic everywhere it's used.

Reply:
I was keeping it with this one because the default should be that it is enabled, and the default boolean is false.

Reply:
Does it need to be initialized by a default boolean value somewhere? I think the default could be made true by changing `TransformationOptions.Default` to have `Enabled = true`. It's a really minor issue so feel free to ignore this though. :)

Reply:
changed to Enable